addons/crm/tests/test_crm_pls.py

```diff
@@ -351,3 +351,39 @@ def test_predictive_lead_scoring(self):
         self.assertEqual(tools.float_compare(lead_tag_1.automated_probability, 33.69, 2), 0)
         self.assertEqual(tools.float_compare(lead_tag_2.automated_probability, 23.51, 2), 0)
         self.assertEqual(tools.float_compare(lead_tag_1_2.automated_probability, 28.05, 2), 0)
+
+        # remove tag_ids from the calculation
+        self.env['ir.config_parameter'].sudo().set_param("crm.pls_fields", "country_id,state_id,email_state,phone_state,source_id")
+        Lead._cron_update_automated_probabilities()
+        leads_with_tags.invalidate_cache()
+
+        self.assertEqual(tools.float_compare(lead_tag_1.automated_probability, 28.6, 2), 0)
+        self.assertEqual(tools.float_compare(lead_tag_2.automated_probability, 28.6, 2), 0)
+        self.assertEqual(tools.float_compare(lead_tag_1_2.automated_probability, 28.6, 2), 0)
+
+        lead_tag_1.tag_ids = [(5, 0, 0)]  # remove all tags
+        lead_tag_2.tag_ids = [(4, tag_ids[0])]  # add tag 1
+        lead_tag_1_2.tag_ids = [(3, tag_ids[1], 0)]  # remove tag 2
+
+        self.assertEqual(tools.float_compare(lead_tag_1.automated_probability, 28.6, 2), 0)
+        self.assertEqual(tools.float_compare(lead_tag_2.automated_probability, 28.6, 2), 0)
+        self.assertEqual(tools.float_compare(lead_tag_1_2.automated_probability, 28.6, 2), 0)
+
+        # remove all pls fields
+        self.assertEqual(tools.float_compare(leads[3].automated_probability, 20.87, 2), 0)
+        self.assertEqual(tools.float_compare(leads[8].automated_probability, 2.43, 2), 0)
+
+        self.env['ir.config_parameter'].sudo().set_param("crm.pls_fields", False)
```

Comment:
In fact, this test does not test the unlink, as the param still exists (set to False, but still). So I'm not sure if unlink method is still necessary.

In fact, once the config parameter is created, it will remains ad vitam eternam, unless the module is uninstalled, in which case we don't really care to align the depends.
So, I think you can remove the unlink override, but still keep those tests, as it's testing the impact of the non-optional PLS fields (stage_id and team_id)